### ISSUE TITLE
‼️ NOTICE: (ec2) VPCs with SubnetType.PRIVATE cause resource destruction on versions >=v1.162.0 & >=1.163.2

### ISSUE THREAD

**Issue by corymhall** [aws/aws-cdk#21138]
### Describe the bug

Please add your +1 👍 to let us know you have encountered this
---

### Overview:

PR https://github.com/aws/aws-cdk/pull/19320 changed the values of duplicate enums. `SubnetType.PRIVATE` was changed from `Private` to `Deprecated_Private`.

This is causing separate issues on v1 vs v2.

**On v1**
For example, a basic VPC

```ts
new ec2.Vpc(this, 'Vpc', {
      subnetConfiguration: [
        {
          name: 'Private',
          subnetType: ec2.SubnetType.PRIVATE,
        },
        {
          name: 'Public',
          subnetType: ec2.SubnetType.PUBLIC,
        },

      ]
    })
```
which is causing resources to be destroyed/recreated.

```
Resources
[-] AWS::EC2::Route VpcPrivateSubnet1DefaultRouteBE02A9ED destroy
[-] AWS::EC2::Route VpcPrivateSubnet2DefaultRoute060D2087 destroy
[-] AWS::EC2::EIP VpcPublicSubnet1EIPD7E02669 destroy
[-] AWS::EC2::NatGateway VpcPublicSubnet1NATGateway4D7517AA destroy
[-] AWS::EC2::EIP VpcPublicSubnet2EIP3C605A87 destroy
[-] AWS::EC2::NatGateway VpcPublicSubnet2NATGateway9182C01D destroy

```

**On v2**
`SubnetType.PRIVATE` has been removed and any code using it will not longer build.

### Solution:

Upgrade to `v1.164.0`

### Related Issues:

https://github.com/aws/aws-cdk/issues/21131

**Comment by corymhall** [comment]
This issue has been fixed in https://github.com/aws/aws-cdk/pull/21140 and will be in the next release `v1.164.0` which are working on releasing.

**Comment by zhaoyuguang** [comment]
> PR #19320 changed the values of duplicate enums. `SubnetType.PRIVATE` was changed from `Private` to `Deprecated_Private`.
Hello. Please describe the rootcause more clearly, I can't understand the code ` PRIVATE = 'Deprecated_Private', it will bring such a big impact,Please tell how to affect the resource generation of VPC, thanks

**Comment by corymhall** [comment]
This issue is fixed in `v1.164.0`.
